Tools/clinic/clinic.py

```diff
@@ -1199,7 +1199,7 @@ def parser_body(
 
         fastcall = not new_or_init
         limited_capi = clinic.limited_capi
-        if limited_capi and (requires_defining_class or pseudo_args or
+        if limited_capi and (pseudo_args or
                 (any(p.is_optional() for p in parameters) and
                  any(p.is_keyword_only() and not p.is_optional() for p in parameters)) or
                 any(c.broken_limited_capi for c in converters)):
```

Comment:
I find this quite hard to read. I'd find it easier to parse if we used a formatting style a little closer to what black might do:

```suggestion
        if limited_capi and (
            pseudo_args
            or (any(p.is_optional() for p in parameters)
                and any(p.is_keyword_only() and not p.is_optional() for p in parameters))
            or any(c.broken_limited_capi for c in converters)
        ):
```

Reply:
I would prefer to have a kind of consistency in terms of coding style. If this block is changed, many other lines should be changed. For example, the style in this file is to put the operator at the end of a line, not at the beginning.

I need this fix to unblock a large following change: PR #110966.

Once I landed these changes, feel free to run Black, or change the coding style manually :-)

Reply:
I only comment about formatting here because I think the poor readability was a contributing factor to why a bug crept in to these lines. But, sure, we can leave this for now :-)

Reply:
I don't think that it was a bug. IMO my change is wrong :-) But I am not sure... AC is complicated and has a small test suite. Maybe recent changes for limited C API already cover all cases. Maybe not.

I made the change because it prevented me to merge PR #110966: with this PR, md5 clinic code started again to use the internal C API which was wrong. I mean, it injected a `pycore_modsupport.h` which is incompatible with the limited C API, even if the include was not used.

Reply:
> But, sure, we can leave this for now :-)

I would suggest to run Black on the whole file, not just one a few lines.